### PR TITLE
upgrade tinymce

### PR DESCRIPTION
**Pull request by xithan**
also:
- fix and refactor bootswatch themes (Everything is compiled in one css theme card now. I had to fork the bootswatch repostiory to change the build process so that the themes include the select2 library. That's necessary because it depends on variables defined in the themes)
- add a theme that is easy to customize

### CODE REVIEW

**Comment by xithan** [comment]
The code organization is a bit odd. Tinymce is a mod in card but it depends on assets in `decko/rails/assets`

**Comment by ethn** [comment]
I was just noticing that, too.

File: card/mod/standard/set/all/rich_html/toolbar.rb [hunk 1]
```diff
@@ -143,8 +143,8 @@ def nests_editable?
   def account_split_button
     return "" unless card.accountable?
     toolbar_split_button "account", related: :account, icon: :account_box do
-      %i[ account roles created edited follow ].each_with_object({}) do |item, hash|
-        label = item == :account ? "details" : item.to_s # TODO: i18n
+      %i[account roles created edited follow].each_with_object({}) do |item, hash|
+        label = item == :account ? tr(:details) : tr(item)
```
Comment (xithan) [inline card/mod/standard/set/all/rich_html/toolbar.rb]: @ethn writing your TODO comment was more work than adding i18n

Comment (ethn) [reply card/mod/standard/set/all/rich_html/toolbar.rb]: ha, I didn't realize the translation was there.

File: card/mod/tinymce_editor/lib/javascript/script_tinymce_config.js.coffee [hunk 2]
```diff
@@ -19,17 +19,20 @@ $.extend decko,
...
+      theme: "modern",
       plugins: 'autoresize'
-      autoresize_max_height: 500
+      autoresize_max_height: 500,
     }
     user_conf = if decko.tinyMCEConfig? then decko.tinyMCEConfig else {}
     hard_conf = {
-      mode: 'exact'
-      elements: el_id
+      selector: "##{el_id}",
```
Comment (ethn) [inline card/mod/tinymce_editor/lib/javascript/script_tinymce_config.js.coffee]: inconsistent commas in this file (here and above).  I think coffeescript convention is no comma.

File: card/mod/tinymce_editor/lib/javascript/script_tinymce_config.js.coffee [hunk 3]
```diff
@@ -19,17 +19,20 @@ $.extend decko,
...
+      branding: false
       # CSS could be made optional, but it may involve migrating old legacy
       # *tinyMCE settings to get rid of stale stuff.
       content_css: decko.cssPath
       entity_encoding: 'raw'
     }
     $.extend conf, user_conf, hard_conf
+    tinyMCE.baseURL = '/assets/tinymce'
```
Comment (ethn) [inline card/mod/tinymce_editor/lib/javascript/script_tinymce_config.js.coffee]: so we're still doing this, eh?  any plans to move away from this approach?

File: decko/rails/assets/tinymce/license.txt [hunk 4]
```diff
@@ -20,7 +20,7 @@ free software--to make sure the software is free for all its users.
   This license, the Lesser General Public License, applies to some
 specially designated software packages--typically libraries--of the
 Free Software Foundation and other authors who decide to use it.  You
-can use it too, but we suggest you first think carefuly about whether
+can use it too, but we suggest you first think carefully about whether
```
Comment (ethn) [inline decko/rails/assets/tinymce/license.txt]: the 20-year typo?

File: card/lib/card/model/save_helper.rb [hunk 5]
```diff
@@ -231,21 +235,23 @@ def add_coderule_item name, prefix, type_id, to
...
+        send "#{method_name}_card", args.merge(type_id: cardtype_card.id)
       end
 
       def respond_to_missing? method, _include_private=false
         extract_cardtype_from_method_name(method) || super
       end
 
       def extract_cardtype_from_method_name method
-        return unless method =~ /^create_(?<type>.+)$/
+        return unless method =~ /^(?<method_name>create|ensure)_(?<type>.+)(?:_card)?$/
```
Comment (ethn) [inline card/lib/card/model/save_helper.rb]: haven't seen these.  var names in regex.  I like.

Comment (xithan) [reply card/lib/card/model/save_helper.rb]: Then you missed several chances in recent years to get to know each other :smile: The last time you passed each other without notice was probably here https://github.com/decko-commons/decko/pull/80/files#diff-be05844104fae4d5b9ea9e1b41dd8040R3

File: card/mod/tinymce_editor/lib/javascript/script_tinymce_config.js.coffee (hunk 3 above)
Comment (xithan) [reply card/mod/tinymce_editor/lib/javascript/script_tinymce_config.js.coffee]: no idea how

Comment (ethn) [reply card/mod/tinymce_editor/lib/javascript/script_tinymce_config.js.coffee]: yeah, me neither.

**Pull request merged**